WebDriverAgentLib/Utilities/XCUIApplicationProcessQuiescence.m

```diff
@@ -8,44 +8,67 @@
  */
 
 #import "XCUIApplicationProcessQuiescence.h"
-#import "XCUIApplicationProcess.h"
+
 #import <objc/runtime.h>
+
+#import "FBConfiguration.h"
 #import "FBLogger.h"
+#import "XCUIApplicationProcess.h"
 
 static void (*original_waitForQuiescenceIncludingAnimationsIdle)(id, SEL, BOOL);
-static BOOL isWaitForQuiescence = NO;
-static BOOL isAnimationCheckEnabled = YES;
 
-@implementation XCUIApplicationProcessQuiescence
 
-+ (void)load
+static void swizzledWaitForQuiescenceIncludingAnimationsIdle(id self, SEL _cmd, BOOL includeAnimations)
 {
-  Method waitForQuiescenceMethod = class_getInstanceMethod([XCUIApplicationProcess class], @selector(waitForQuiescenceIncludingAnimationsIdle:));
-  if (waitForQuiescenceMethod != nil) {
-    original_waitForQuiescenceIncludingAnimationsIdle = (void(*)(id, SEL, BOOL)) method_getImplementation(waitForQuiescenceMethod);
-    Method newMethod = class_getClassMethod([XCUIApplicationProcessQuiescence class], @selector(swizzledWaitForQuiescenceIncludingAnimationsIdle:));
-    method_setImplementation(waitForQuiescenceMethod, method_getImplementation(newMethod));
-  } else {
-    [FBLogger log:@"Could not find method -[XCUIApplicationProcess waitForQuiescenceIncludingAnimationsIdle:]"];
+  if (!FBConfiguration.shouldWaitForQuiescence || FBConfiguration.waitForIdleTimeout < DBL_EPSILON) {
+    return;
   }
-}
+  original_waitForQuiescenceIncludingAnimationsIdle(self, _cmd, includeAnimations);
```

Comment:
Shouldn't we refrain from calling the original implementation if everything is handled in this function?
Calling it will trigger XCTest's checks, which can hang sometimes.

Reply:
yes, I can change the implementation to a completely custom one